### PR TITLE
Njavokhir/client auth error

### LINKED ISSUE #43: Client able to access client portal without any authorized token

### PR DESCRIPTION
**Pull request by NJavokhir**
### SECTION 1: RESOURCE SERVER (http://localhost:5001/)
- [x] http://localhost:5001/ home page loads
- [x] "LoginAsUser" button takes you to the user login page
- [x] Log in using "tester" and "12341234" should succeed
- [x] Updating the display name should work
- [x] Log out and Register a new user and try to log in with that new user
- [x] "LoginAsClient" button takes you to the client login page
- [x] Log in using "layer8" and "12341234" should work
- [x] You will be able to see your UID and Secret in profile
- [x] Log out and Register a new client and log in using that should work

### SECTION 2: WE'VE GOT POEMS (http://localhost:5173/)
- [ ] Log in with 'tester' / '1234'
- [ ] Log in anonymously with Layer8
- [ ] Clicking "Get Next Poem" loads different poem correctly x 3
- [ ] Clicking "Logout" takes you to the login screen
- [ ] Clicking "Register" takes you to the registration page
- [ ] Registering with a username, password, and profile image is successful
- [ ] Logging in with the new username / password succeeds
- [ ] Logging in with Layer8 opens the pop-up
- [ ] Logging in with the newly registered credentials from Section 1 succeeds
- [ ] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
- [ ] Clicking "Get Next Poem" loads different poem correctly x 3
- [ ] Clicking "Logout" takes you to the login page
- [ ] Log in with 'tester' / '1234', and then with the credentials from Section 1 succeeds
- [ ] This time, DO NOT share "display name" or "country"
- [ ] Clicking "Get Next Poem" loads different poem correctly x 3

### SECTION 3: IMSHARER (http://localhost:5174/)
- [ ] Main page loads
- [ ] Upload of image works
- [ ] Reload leads to instant reload (demonstrating proper caching)
- [ ] Clicking the newly loaded image shows it in a lightbox

### Client Aauth Error

**Issue**: Clients were able to access the client portal without an authorized token, posing a security risk. This was reported in issue #43.

**Solution**:
- Added error handling to redirect unauthorized access attempts to the login page with an appropriate error message.

**Impact**:
- This fix enhances the security of the client portal by preventing unauthorized access.
- No changes to the database schema or external APIs.
- Minimal impact on performance as token validation is a lightweight check.

**Related Issues**:
- Closes #43

### CODE REVIEW

**Comment by NJavokhir** [comment]
@ternakkode This PR is ready to review.

File: server/assets-v1/templates/src/pages/client_portal/register.html [hunk 1]
```diff
@@ -230,6 +221,14 @@ <h1 class="font-bold text-3xl md:text-4xl text-[#3751FE] mb-11 animate-slideFrom
           return false;
         }
       };
+      const showToastMessage = (message, type) => {
```
Comment (stravid87) [inline server/assets-v1/templates/src/pages/client_portal/register.html:224]: I notice here that you use the same function `showToastMessage` as you do in the profile.html page. Not for this PR, but I wonder if we should explore putting this message in a simple library of utils. Notice that on line 152, you import such things as `ref` from Vue. I know we're using a CDN, but I wonder could we still have our own libraries?

**Review by stravid87 — CHANGES_REQUESTED** [review]
Straight forward, single responsibility PR. Nice work. I appreciate the explanatory comment as well.  
No changes required.
Good work.

**Comment by stravid87** [comment]
Also, please complete manual testing on Section 1 of the manual tests above. 

```
 SECTION 1: RESOURCE SERVER (http://localhost:5001/)

...
```

**Review by stravid87 — APPROVED** [review]

**Comment by ternakkode** [comment]
I dont think this ticket solve the issue, try to reproducte it by opening the https://localhost:5001/client-profile without login first, the expected behavior is the user should redirected to login page instead of showing the blank screen

**Review by ternakkode — CHANGES_REQUESTED** [review]
I don't think this ticket solves the issue. Try reproducing it by opening https://localhost:5001/client-profile without logging in first. The expected behavior is that the user should be redirected to the login page instead of seeing a blank screen.

**Comment by NJavokhir** [comment]
> I don't think this ticket solves the issue. Try reproducing it by opening https://localhost:5001/client-profile without logging in first. The expected behavior is that the user should be redirected to the login page instead of seeing a blank screen.

Yes, you are right. I solved this problem on my last changes.

**Review by ternakkode — APPROVED** [review]

**Comment by NJavokhir** [comment]
> Also, please complete manual testing on Section 1 of the manual tests above and then consider any of Firhans suggestions. Otherwise, I'm good.
> 
> ```
>  SECTION 1: RESOURCE SERVER (http://localhost:5001/)
> 
> ...
> ```

Done!

File: server/assets-v1/templates/src/pages/client_portal/register.html (hunk 1 above)
Comment (NJavokhir) [reply server/assets-v1/templates/src/pages/client_portal/register.html:224]: Yes, we can import showToastMessage function from utils folder. We can implement it. Maybe, I will create extra issue for this.

**Pull request merged**